web/src/js/components/Modes/Popover.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from "react";
+
+interface PopoverProps {
+    children: React.ReactNode;
+}
+
+export function Popover({ children }: PopoverProps) {
+    const [isVisible, setIsVisible] = React.useState(false);
+    const popoverRef = React.useRef<HTMLDivElement>(null);
+
+    const handleClickOutside = (event: MouseEvent) => {
+        if (
+            popoverRef.current &&
+            !popoverRef.current.contains(event.target as Node)
+        ) {
+            setIsVisible(false);
+        }
+    };
+
+    React.useEffect(() => {
+        document.addEventListener("mousedown", handleClickOutside);
+        return () => {
+            document.removeEventListener("mousedown", handleClickOutside);
+        };
+    }, []);
```

Comment:
okay, yes sure :)

Reply:
I'm not sure it's already available! It looks like it will be supported with react 19 : https://stackoverflow.com/questions/78203051/is-it-possible-to-use-the-html-popover-attribute-in-a-react-element

Reply:
Okay, I've updated the logic using the popover API, but I have two concerns:

1. The `Anchor Positioning` feature isn't supported across all browsers. Currently, it only works in Chrome and Microsoft Edge (see [here](https://developer.chrome.com/blog/anchor-positioning-api) and [here](https://developer.chrome.com/blog/introducing-popover-api#anchor_positioning) for details).
   
2. I haven’t noticed any clear advantages in moving to this new version. I think that the previous implementation was simpler and more intuitive. This new version introduces additional complexity in the CSS. What do you think? Maybe it's just a personal preference :)

PS: The issue where the popover closes automatically after a successful dispatch event and correctly set value persists (see screen recording).